src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -116,18 +117,29 @@ public Launcher decorate(final Launcher launcher, final Node node) {
             public Proc launch(ProcStarter starter) throws IOException {
                 boolean quiet = starter.quiet();
                 FilePath pwd = starter.pwd();
-                String[] commands = getCommands(starter);
 
-                return doLaunch(quiet, pwd, commands);
+                String [] cmdEnvs = starter.envs();
+
+                //check if the cmd is sourced from Jenkins, rather than another plugin; if so, skip cmdEnvs as we are getting other environment variables
+                for (String cmd : cmdEnvs) {
+                    if (cmd.startsWith(JENKINS_HOME)) {
```

Comment:
why was this introduced? It is preventing environment variable injecting wrappers to inject env into sh steps.

Reply:
From #232 - "One issue that cropped up was that when executing sh commands, we would get the jnlp agent's injected environment variables as well, causing obvious problems such as JAVA_HOME being overwritten. The unsatisfying answer was to check for the presence of JENKINS_HOME in the cmdenvs and skip if present."

Reply:
might this help in this situation?
https://github.com/jenkinsci/kubernetes-plugin/pull/242/files#diff-8a1ab34452b01252e89ffeee0b2eda45R278